service/migration_manager.cc

```diff
@@ -851,10 +852,10 @@ future<std::vector<mutation>> prepare_new_view_announcement(storage_proxy& sp, v
             return include_keyspace(sp, *keyspace, std::move(mutations)).get();
         });
     } catch (const replica::no_such_keyspace& e) {
-        return make_exception_future<std::vector<mutation>>(
-            exceptions::configuration_exception(format("Cannot add view '{}' to non existing keyspace '{}'.", view->cf_name(), view->ks_name())));
+        auto&& ex = std::make_exception_ptr(exceptions::configuration_exception(format("Cannot add view '{}' to non existing keyspace '{}'.",
```

Comment:
This is either confusing, or I just don't know C++ well - but why did you use `auto&& ex = <rvalue>`? Does this work at all? My first thought was that this will immediately create a dangling reference.